### PR TITLE
Typechain Polkadot Milestone 1

### PR DESCRIPTION
**Pull request by coreggon11**
# Milestone Delivery Checklist

- [x] The [milestone-delivery-template.md](https://github.com/w3f/Grant-Milestone-Delivery/blob/master/deliveries/milestone-delivery-template.md) has been copied and updated.
- [X] The [invoice form :pencil:](https://forms.gle/LSRr7PCjBpEbKGh89) has been filled out for this milestone.
- [X] This pull request is being made by the same account as the accepted application.
- [X] In case of acceptance, the payment will be transferred to the BTC/ETH payment address provided in the application.
- [X] The delivery is according to the [Guidelines for Milestone Deliverables](https://github.com/w3f/Grants-Program/blob/master/docs/Support%20Docs/milestone-deliverables-guidelines.md).

Link to the application pull request: https://github.com/w3f/Grants-Program/blob/master/applications/typechain-polkadot-follow-up-2.md

### CODE REVIEW

File: deliveries/typechain-polkadot-follow-up-2.md [hunk 1]
```diff
@@ -0,0 +1,42 @@
+# Milestone Delivery :mailbox:
+
+**The [invoice form :pencil:](https://docs.google.com/forms/d/e/1FAIpQLSfmNYaoCgrxyhzgoKQ0ynQvnNRoTmgApz9NrMp-hd8mhIiO0A/viewform) has been filled out correctly for this milestone and the delivery is according to the official [milestone delivery guidelines](https://github.com/w3f/Grants-Program/blob/master/docs/Support%20Docs/milestone-deliverables-guidelines.md).**
+
+* **Application Document:** https://github.com/w3f/Grants-Program/pull/1432
```
Comment (0xCaso) [inline deliveries/typechain-polkadot-follow-up-2.md:5]: Fix the **Application Document** here:
```suggestion
* **Application Document:** https://github.com/w3f/Grants-Program/blob/master/applications/typechain-polkadot-follow-up-2.md
```

**Review by 0xCaso — COMMENTED** [review]
Hi @coreggon11, thanks for the new delivery! We'll get back to you as soon as possible.

In the meantime, could you do this little edit to the delivery document?

File: deliveries/typechain-polkadot-follow-up-2.md (hunk 1 above)
Comment (coreggon11) [reply deliveries/typechain-polkadot-follow-up-2.md:5]: sure, thanks for pointing this out

**Comment by varex83** [comment]
Hello, @0xCaso do you have any updates regarding our delivery?

**Comment by 0xCaso** [comment]
Apologies for the delay, currently we have some backlog. If no one will start the evaluation today, for sure it will be done during the start of the next week.

**Comment by varex83** [comment]
Hello @0xCaso , yes

**Review by 0xCaso — COMMENTED** [review]
I was trying the guide for the `typechain-compiler` ([this](https://github.com/727-Ventures/typechain-polkadot/blob/master/docs/about.md) one) and when I try to use the compiler, I get the following error:
```
======== Found contracts ========
	flipper,
	my_psp22
======== Compiling flipper ========
ERROR: Mismatching versions of `parity-scale-codec` were found!
Please ensure that your contract and your ink! dependencies use a compatible version of this package.
Error: Command failed: cargo +nightly contract build --manifest-path ./contracts/flipper/Cargo.toml
ERROR: Mismatching versions of `parity-scale-codec` were found!
Please ensure that your contract and your ink! dependencies use a compatible version of this package.
```
Maybe there is something wrong in the `Cargo.toml` files?

Also, I had to edit the `package.json` file, updating `"@polkadot/keyring": "^9.13.4"` to `"@polkadot/keyring": "^10.4.2"`

**Comment by 0xCaso** [comment]
Ok, I updated cargo-contract and now I get a different error:
```
Compiling flipper v0.1.0 (/private/var/folders/9_/h5h1hjb11qzddpjd7kr1sn380000gn/T/cargo-contract_2sbo1C)
   Compiling metadata-gen v0.1.0 (/private/var/folders/9_/h5h1hjb11qzddpjd7kr1sn380000gn/T/cargo-contract_2sbo1C/.ink/metadata_gen)
    Finished release [optimized] target(s) in 22.97s
     Running `target/ink/release/metadata-gen ''`
Error: ENOENT: no such file or directory, open '/GitHub/typechain-compiler-example/contracts/flipper/target/ink/metadata.json'
    at Object.openSync (node:fs:600:3)
    at Object.readFileSync (node:fs:468:35)
    at main (/GitHub/typechain-compiler-example/node_modules/@727-ventures/typechain-compiler/index.ts:114:11)
    at Object.<anonymous> (/GitHub/typechain-compiler-example/node_modules/@727-ventures/typechain-compiler/index.ts:143:1)
    at Module._compile (node:internal/modules/cjs/loader:1159:14)
    at Module.m._compile (/GitHub/typechain-compiler-example/node_modules/ts-node/src/index.ts:1618:23)
    at Module._extensions..js (node:internal/modules/cjs/loader:1213:10)
    at Object.require.extensions.<computed> [as .ts] (/GitHub/typechain-compiler-example/node_modules/ts-node/src/index.ts:1621:12)
    at Module.load (node:internal/modules/cjs/loader:1037:32)
    at Function.Module._load (node:internal/modules/cjs/loader:878:12) {
  errno: -2,
  syscall: 'open',
  code: 'ENOENT',
  path: '/GitHub/typechain-compiler-example/contracts/flipper/target/ink/metadata.json'
}
```

**Comment by varex83** [comment]
Hello @0xCaso, thanks for the quick response, do you use CARGO_TARGET_DIR? Also can you check if `target` directory located in the` ./contracts/flipper`

**Comment by varex83** [comment]
@0xCaso try to use `@727-ventures/typechain-compiler@0.5.16`

**Comment by 0xCaso** [comment]
Thanks a lot for the super fast replies and update, now it works perfectly!
I'll update you tomorrow :)

**Review by 0xCaso — CHANGES_REQUESTED** [review]
May you fix in the guide and in the article the following things:
- The `keyring` correct version that I mentioned

  > Also, I had to edit the package.json file, updating "@polkadot/keyring": "^9.13.4" to "@polkadot/keyring": "^10.4.2"
- `npx @727-Ventures/typechain-compiler --config typechain.config.json` has `@727-Ventures` instead of `@727-ventures`, which makes the command fail
- Specify if it's needed to use `@727-ventures/typechain-compiler@0.5.16`, or  `@727-ventures/typechain-compiler` is sufficient

**Comment by varex83** [comment]
Hello, @0xCaso okay, will be done in a few minutes

**Comment by varex83** [comment]
@0xCaso done ✅

**Review by 0xCaso — APPROVED** [review]
Thanks a lot again for the quick fix! I'm approving the milestone :)

**Comment by varex83** [comment]
Thank you too @0xCaso for the quick review and replies! Appreciate that!

**Comment by 0xCaso** [comment]
> Thank you too @0xCaso for the quick review and replies! Appreciate that!

Thanks!

Did you send the invoice for this milestone?

**Comment by 0xCaso** [comment]
I found it, but seems like you've resent an old one (Sol2Ink milestone 2 grant) on Feb 13

**Comment by coreggon11** [comment]
Hi @0xCaso that is my fault, I probably mixed it up. I will fix it ASAP.

**Comment by coreggon11** [comment]
Fixed. Sorry for incovenience

**Comment by coreggon11** [comment]
Hi @0xCaso the new invoice was submitted.

**Comment by 0xCaso** [comment]
Thanks a lot, I forwarded it internally. The payment should happen within 14 days!

**Comment by 0xCaso** [comment]
Hi @coreggon11, apologies for the delay. The payment should happen tomorrow.

**Comment by RouvenP** [comment]
hi @coreggon11 we just transferred the payment. Apologies for the delay!

**Pull request merged**